commands/MessageCollector.go

```diff
@@ -0,0 +1,42 @@
+package commands
+
+import (
+	"errors"
+	"github.com/bwmarrin/discordgo"
+	"strconv"
+)
+
+/**
+* /collect <message-snowflake-id> <condition-type> <condition-value>
+* Condition type: number, time
+* Condition value: # Messages to collect starting with ID or timestamp to collect till
+*/
+type MessageCollector struct{}
+
+
+func (h MessageCollector) Name() string {
+	return "Message Collector"
+}
+
+func (h MessageCollector) Description() string {
+	return "Collects messages starting from a given message ID to a given limit to local storage for further processing"
+}
+
+func (h MessageCollector) Execute(s *discordgo.Session, m *discordgo.MessageCreate, cmdSlices []string) (bool, error) {
+	if cmdSlices[0] != "collect" {
```

Comment:
`cmdSlices` could be empty (I know it never will be in the way you plan to use it, but it never hurts to add bounds checking).

Reply:
Did bound check in the message processor, it shouldn't really see if the message could be handled by any of the commands if its an empty string.